core/src/apps/webauthn/__init__.py

```diff
@@ -497,20 +497,15 @@ def __call__(self) -> None:
         send_cmd_sync(cmd_keepalive(self.cid, _KEEPALIVE_STATUS_PROCESSING), self.iface)
 
 
-async def check_pin(keepalive_callback: KeepaliveCallback) -> bool:
-    from apps.common.request_pin import PinCancelled, request_pin
+async def verify_user(keepalive_callback: KeepaliveCallback) -> bool:
+    from apps.common.request_pin import verify_user_pin
     import trezor.pin
 
     try:
         trezor.pin.keepalive_callback = keepalive_callback
-        if config.has_pin():
-            pin = await request_pin("Enter your PIN", config.get_pin_rem())
-            while config.unlock(trezor.pin.pin_to_int(pin)) is not True:
-                pin = await request_pin("Wrong PIN, enter again", config.get_pin_rem())
-            ret = True
-        else:
-            ret = config.unlock(trezor.pin.pin_to_int(""))
-    except PinCancelled:
+        await verify_user_pin()
+        ret = True
+    except Exception:
```

Comment:
We should never use `except Exception` and try to enumerate which exception classes we are trying to catch here.

Reply:
Changed to catch PinInvalid and PinCancelled. The RuntimeError implies some major inconsistency, so I am letting it propagate.
https://github.com/trezor/trezor-firmware/pull/526/commits/5a7e2fdb7a4fa87372bbd246eddfb4769b9da8b8